### PR TITLE
Support loading options from an INI file

### PR DESCRIPTION
**Pull request by ccawley2011**
This replaces the old code for reading ST-506 parameters with a more flexible version based on [inih](https://github.com/benhoyt/inih). This allows defaults for most of the command line parameters to be specified in the INI file, and potentially more things in the future. The INI file (and the hexcmos file) is stored in the Application Data directory on Windows and in `XDG_DATA_HOME` on Linux, but will default to the current directory if those can't be found.

Since this PR is big, changes things for the user, and would be hard to change again later, I'd appreciate it if I could get a second opinion on it before it's merged.

### CODE REVIEW

**Comment by chris-y** [comment]
I haven't had a proper look at this yet, but is XDG_DATA_HOME the right place?  On this Pi I'm using currently that doesn't exist.  XDG_CONFIG_HOME does and is probably better?

**Comment by chris-y** [comment]
Can you add an example/default ini file?

**Comment by ccawley2011** [comment]
> I haven't had a proper look at this yet, but is XDG_DATA_HOME the right place? On this Pi I'm using currently that doesn't exist. XDG_CONFIG_HOME does and is probably better?

I think `XDG_CONFIG_HOME` is technically better, but `XDG_DATA_HOME` is what SDL2 uses to implement `SDL_GetPrefPath()`, and the X11 port matches the SDL port for convenience. There's some additional detail here: libsdl-org/SDL#4447

> Can you add an example/default ini file?

Here's an example of what `arcem.ini` looks like. It probably doesn't make much sense to check it in here, but I'll have a go at integrating it into the manual.
```
[machine]
rom=ROM
hostfsdir=D:\Projects\ArcEm\hostfs
processor=ARM3
memory=16M

[mfm1]
path=D:\Projects\ArcEm\HardImage1
cylinders=1024
heads=8
sectors=32
reclength=256

[fdc0]
path=X:\CoverDiscs\AcornUser\AUMay94.adf

[fdc1]
path=X:\CoverDiscs\AcornUser\AUMay95.adf
```

File: arch/ArcemConfig.c [hunk 1]
```diff
@@ -228,47 +342,9 @@ void ArcemConfig_ParseCommandLine(int argc, char *argv[])
...
-          iArgument += 2;
-        }
-        else if(0 == strcmp("16M", argv[iArgument + 1])) {
-          hArcemConfig.eMemSize = MemSize_16M;
-
-          iArgument += 2;
-        }
-        else {
+        if (arcemconfig_StringToEnum(&uValue, argv[iArgument + 1], memsize_labels)) {
+          pConfig->eMemSize = uValue;
```
Comment (Phlamethrower) [inline arch/ArcemConfig.c:346]: `iArgument += 2` needed to avoid infinite loop

File: arch/ArcemConfig.c [hunk 2]
```diff
@@ -279,22 +355,9 @@ void ArcemConfig_ParseCommandLine(int argc, char *argv[])
...
-          iArgument += 2;
-        }
-        else if(0 == strcmp("ARM3", argv[iArgument + 1])) {
-          hArcemConfig.eProcessor = Processor_ARM3;
-
-          iArgument += 2;
-        }
-        else {
+        if (arcemconfig_StringToEnum(&uValue, argv[iArgument + 1], processor_labels)) {
+          pConfig->eProcessor = uValue;
```
Comment (Phlamethrower) [inline arch/ArcemConfig.c:359]: `iArgument += 2` needed to avoid infinite loop

**Review by Phlamethrower — CHANGES_REQUESTED** [review]
Looks good, and the INI file seems to work OK with a quick test of the RISC OS version.

This set of changes will make the RISC OS !Help file out-of-date, but rather than updating that help file maybe it'd be better to change it to use the HTML manual as the !Help file instead.

Do you need anyone to help remove the hArcemConfig references from the RISC OS sources? I don't currently have a GCCSDK install set up but it's probably about time I got it working again.

**Comment by ccawley2011** [comment]
Thanks, I've addressed the review comments.

> Do you need anyone to help remove the hArcemConfig references from the RISC OS sources?

If you could help with that, that'll be great.

**Review by Phlamethrower — APPROVED** [review]

**Comment by chris-y** [comment]
I tested on the Amiga build, but as I'm using tooltypes I didn't really check the ini file loading! The tooltypes still work though, the ini is more of a backup for advanced config.

**Pull request merged**